src/main/java/org/folio/inventory/services/InstanceIdStorageService.java

```diff
@@ -0,0 +1,26 @@
+package org.folio.inventory.services;
+
+import io.vertx.core.Future;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.folio.inventory.common.dao.EntityIdStorageDao;
+import org.folio.inventory.domain.relationship.EntityTable;
+import org.folio.inventory.domain.relationship.RecordToEntity;
+
+public class InstanceIdStorageService implements IdStorageService {
+  private static final Logger LOGGER = LogManager.getLogger(InstanceIdStorageService.class);
+
+  private final EntityIdStorageDao entityIdStorageDao;
+
+  public InstanceIdStorageService(EntityIdStorageDao entityIdStorageDao) {
+    this.entityIdStorageDao = entityIdStorageDao;
+  }
+
+  @Override
+  public Future<RecordToEntity> store(String recordId, String instanceId, String tenantId) {
+    RecordToEntity recordToInstance = RecordToEntity.builder().table(EntityTable.INSTANCE).recordId(recordId).entityId(instanceId).build();
+    LOGGER.info("Saving RecordToInstance relationship: {}", recordToInstance);
+    return entityIdStorageDao.saveRecordToEntityRelationship(recordToInstance, tenantId)
+      .map(recordToInstance);
```

Comment:
Hi @AivarIUsupov, propose to remove map() method call in order to return existing instance id from DB instead of new instanceId that we pass to the method in case of instance id already exists for specified recordId (duplicated event processing)
```suggestion
    return entityIdStorageDao.saveRecordToEntityRelationship(recordToInstance, tenantId);
```

Reply:
Done